tests/std/tests/P0053R7_cpp_synchronized_buffered_ostream/test.hpp

```diff
@@ -0,0 +1,194 @@
+#include <assert.h>
+#include <memory>
+#include <string>
+#include <syncstream>
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+
+constexpr size_t _Min_size_allocation = 50;
+constexpr size_t _Min_syncbuf_size    = 32;
+
+template <class Ty, bool ThrowOnSync = false>
+class string_buffer : public basic_streambuf<Ty, char_traits<Ty>> { // represents the wrapped object in syncbuf
+public:
+    string_buffer()  = default;
+    ~string_buffer() = default;
+
+    streamsize xsputn(const Ty* _Ptr, streamsize _Count) override {
+        str.append(_Ptr, static_cast<string::size_type>(_Count));
+        return _Count;
+    }
+
+    int sync() override {
+        if constexpr (ThrowOnSync) {
+            return -1;
+        } else {
+            return 0;
+        }
+    }
+
+    string str;
+};
+
+class small_size_allocation {
+public:
+    using size_type = size_t;
+
+    _NODISCARD size_type max_size() const noexcept {
+        return _Min_size_allocation;
+    }
+};
+
+template <class Ty>
+class allocator_base {
+public:
+    using value_type = Ty;
+    using pointer    = Ty*;
+
+    _NODISCARD __declspec(allocator) pointer allocate(_CRT_GUARDOVERFLOW const size_t _Count) {
+        return static_cast<pointer>(allocator<value_type>{}.allocate(_Count));
+    }
+
+    void deallocate(pointer const _Ptr, const size_t _Count) noexcept {
+        allocator<value_type>{}.deallocate(_Ptr, _Count);
+    }
+};
+
+template <class Ty>
+class fancy_ptr {

```

Comment:
This is another test-only issue that I don't think should block merging.

Reply:
If anyone wants to fix this before merging, `VSO_0000000_fancy_pointers` has a conformant fancy pointer and allocator. I think it would be sufficient to add instantiation coverage to that test (it's compile-only) and remove the coverage here, since the intersection of "fancy pointer and small size allocation" seems like it should work automatically (which is why we're usually satisfied with compile-only coverage for fancy pointers).

Reply:
I can do that this morning. I'll remove fancy pointer coverage from here, and add `basic_syncbuf` and `basic_osyncstream` to `VSO_0000000_fancy_pointers`